syntax = "proto3";

message FaceAlignmentHeader {
    repeated BoundingBox   source_bboxes = 1;
    // DLib can't currently allow arbitrary alignments
    //repeated FaceLandmarks source = 2;
    //FaceLandmarks target = 3;
}

service AllInOne {
	rpc AllInOne(AllInOneRequest) returns (AllInOneResponse){}
	//rpc AlignFace(stream FaceAlignmentRequest) returns (stream FaceAlignmentResponse){}
}

message AllInOneRequest {
	string image = 1;
	string FileType = 2;
}

message AllInOneResponse {
	repeated string gender = 1;
	repeated string smile = 2;
	repeated int32 Age = 3;
	repeated BoundingBox b_box = 4;
}

message FaceAlignmentRequest {
    oneof header_or_chunk {
        FaceAlignmentHeader header = 1;
        ImageRGB image_chunk = 2;
    }
}

message FaceAlignmentResponseHeader {

}

message FaceAlignmentResponse {
    oneof header_or_chunk {
        FaceAlignmentResponseHeader header = 1;
        ImageRGB image_chunk = 2;
    }
}

message ImageRGB {
	string content = 1;
}

message BoundingBox {
	int32 x = 1;
	int32 y = 2;
	int32 w = 3;
	int32 h = 4;
}

message Point2D {
	int32 x = 1;
	int32 y = 2;
}

message FaceDetections {
	repeated BoundingBox face_bbox = 1;
}

message FaceLandmarks {
	string landmark_model = 1;
	repeated Point2D point = 2;
}

message FaceLandmarkDescriptions {
	string landmark_model = 1;
	repeated string landmark_description = 2;
	repeated Point2D landmark_avg = 3;
}

message FaceLandmarkModels {
	repeated FaceLandmarkDescriptions model = 1;
}


